linera-execution/tests/contract_runtime_apis.rs

```diff
@@ -0,0 +1,829 @@
+    let mut view = SystemExecutionState {
+        description: Some(ChainDescription::Root(0)),
+        ..SystemExecutionState::default()
+    }
+    .into_view()
+    .await;
+
+    let (application_id, application) = view.register_mock_application().await.unwrap();
+
+    application.expect_call(ExpectedCall::execute_operation(
+        move |runtime, _context, _operation| {
+            assert_eq!(
+                runtime.read_owner_balance(missing_account).unwrap(),
+                Amount::ZERO
+            );
+            Ok(vec![])
+        },
+    ));
+    application.expect_call(ExpectedCall::default_finalize());
+
+    let context = create_dummy_operation_context();
+    let mut controller = ResourceController::default();
+    let operation = Operation::User {
+        application_id,
+        bytes: vec![],
+    };
+
+    view.execute_operation(
+        context,
+        Timestamp::from(0),
+        operation,
+        &mut TransactionTracker::new(0, Some(Vec::new())),
+        &mut controller,
+    )
+    .await
+    .unwrap();
+}
+
+/// Tests the contract system API to read all account balances.
+#[proptest(async = "tokio")]
+async fn test_read_owner_balances_system_api(
+    #[strategy(test_accounts_strategy())] accounts: BTreeMap<AccountOwner, Amount>,
+) {
+    let mut view = SystemExecutionState {
+        description: Some(ChainDescription::Root(0)),
+        balances: accounts.clone(),
+        ..SystemExecutionState::default()
+    }
+    .into_view()
+    .await;
+
+    let (application_id, application) = view.register_mock_application().await.unwrap();
+
+    application.expect_call(ExpectedCall::execute_operation(
+        move |runtime, _context, _operation| {
+            assert_eq!(
+                runtime
+                    .read_owner_balances()
+                    .unwrap()
+                    .into_iter()
+                    .collect::<BTreeMap<_, _>>(),
+                accounts
+            );
+            Ok(vec![])
+        },
+    ));
+    application.expect_call(ExpectedCall::default_finalize());
+
+    let context = create_dummy_operation_context();
+    let mut controller = ResourceController::default();
+    let operation = Operation::User {
+        application_id,
+        bytes: vec![],
+    };
+
+    view.execute_operation(
+        context,
+        Timestamp::from(0),
+        operation,
+        &mut TransactionTracker::new(0, Some(Vec::new())),
+        &mut controller,
+    )
+    .await
+    .unwrap();
+}
+
+/// Tests the contract system API to read all account owners.
+#[proptest(async = "tokio")]
+async fn test_read_balance_owners_system_api(
+    #[strategy(test_accounts_strategy())] accounts: BTreeMap<AccountOwner, Amount>,
+) {
+    let mut view = SystemExecutionState {
+        description: Some(ChainDescription::Root(0)),
+        balances: accounts.clone(),
+        ..SystemExecutionState::default()
+    }
+    .into_view()
+    .await;
```

Comment:
I feel like a few test helpers would make this file easier to read and less repetitive. E.g.:

```rust
    let mut view = SystemExecutionState::with_accounts(&accounts).into_view().await;
```

And, with the dummy context, default controller, and tracker with empty oracle list:

```rust
    execute_operation(&mut view, operation).await;
```

Or even:

```rust
    execute_empty_operation(&mut view, application_id).await;
```

Reply:
Agreed, opened #2981 and #2892.